quixstreams/sinks/community/redis.py

```diff
@@ -0,0 +1,80 @@
+import json
+import logging
+import time
+from typing import Any, Callable, Optional, Union
+
+try:
+    import redis
+except ImportError as exc:
+    raise ImportError(
+        f"Package {exc.name} is missing: "
+        'run "pip install quixstreams[redis]" to use RedisSink'
+    ) from exc
+
+from quixstreams.sinks import BatchingSink, SinkBatch
+
+__all__ = ("RedisSink",)
+
+logger = logging.getLogger(__name__)
+
+
+class RedisSink(BatchingSink):
+    def __init__(
+        self,
+        host: str,
+        port: int,
+        db: int,
+        value_serializer: Callable[[Any], Union[bytes, str]] = json.dumps,
+        key_serializer: Optional[Callable[[Any, Any], Union[bytes, str]]] = None,
+        password: Optional[str] = None,
+        socket_timeout: float = 30.0,
+        **kwargs,
+    ):
+        """
+        A connector to sink processed data to Redis.
+        It batches the processed records in memory per topic partition, and flushes them to Redis at the checkpoint.
+
+        :param host: Redis host.
+        :param port: Redis port.
+        :param db: Redis DB number.
+        :param value_serializer: a callable to serialize the value to string or bytes
+            (defaults to json.dumps).
+        :param key_serializer: an optional callable to serialize the key to string or bytes.
+            If not provided, the Kafka message key will be used as is.
+        :param password: Redis password, optional.
+        :param socket_timeout: Redis socket timeout.
+            Default - 30s.
+        :param kwargs: Additional keyword arguments passed to the `redis.Redis` instance.
+        """
+
+        super().__init__()
+        self._redis_uri = f"{host}:{port}/{db}"
+        self._client = redis.Redis(
+            host=host,
+            port=port,
+            db=db,
+            password=password,
+            socket_timeout=socket_timeout,
+            **kwargs,
+        )
+        self._key_serializer = key_serializer
+        self._value_serializer = value_serializer
+
+    def write(self, batch: SinkBatch):
+        # Update keys in Redis using transactions
+        # This way, all update will be executed atomically
+        start = time.monotonic()
+        with self._client.pipeline(transaction=True) as pipe:
+            for item in batch:
+                key = item.key
+                if self._key_serializer is not None:
```

Comment:
I saw the discussion you and Tomas had about the keys, but I'm not sure I understood the intention.

Is there any reason not to just set the key to the kafka key by default (with the ability to change it of course)? 

I understand it would only use/keep the latest value, but that's kind of what I would expect for a key/value store. Seems like that should be the default behavior to me?

Reply:
That's what is happening. If no `key_serializer` is configured you get the message key as the key.

Reply:
Yes, exactly.
If there's no `key_serializer`, the Kafka message key will be used (as it says in the docstring)
Otherwise, the return value of `key_serializer(key, value)` is used.